tests/2_functional-tests.js

```diff
@@ -1,270 +1,160 @@
-            // Test the status and the text response (see the example above). 
-            // Please follow the order -status, -text. We rely on that in our tests.
-            // It should respond 'Hello Guest'
+
+      test("Test GET /hello with no name", function (done) {
+        chai
+          .request(server)
+          .get("/hello")
+          .end(function (err, res) {
             assert.fail(res.status, 200);
-            assert.fail(res.text, 'hello Guest');
-            done();   // Always call the 'done()' callback when finished.
+            assert.fail(res.text, "hello Guest");
+            done();
           });
       });
 
-      /**  Another one... **/
-      test('Test GET /hello with your name',  function(done){ // Don't forget the callback...
-         chai.request(server)             // 'server' is the Express App
-          .get('/hello?name=xy_z') /** <=== Put your name in the query **/ 
-          .end(function(err, res){        // res is the response object
-          
-            // Your tests here.
-            // Replace assert.fail(). Make the test pass.
-            // Test the status and the text response. Follow the test order like above.
+      test("Test GET /hello with your name", function (done) {
+        chai
+          .request(server)
+          .get("/hello?name=xy_z")
+          .end(function (err, res) {
             assert.fail(res.status, 200);
-             assert.fail(res.text, 'hello xy_z'/** <==  Put your name here **/);
-            done();   // Always call the 'done()' callback when finished.
+            assert.fail(res.text, "hello xy_z");
+            done();
           });
       });
-
     });
 
-    // In the next example we'll see how to send data in a request payload (body).
-    // We are going to test a PUT request. The '/travellers' endpoint accepts
-    // a JSON object taking the structure :
-    // {surname: [last name of a traveller of the past]} ,
-    // The endpoint responds with :
-    // {name: [first name], surname:[last name], dates: [birth - death years]}
-    // see the server code for more details.
-    
-    // ### EXAMPLE ### 
-    suite('PUT /travellers', function(){
-      test('#example - responds with appropriate JSON data when sending {surname: "Polo"}',  function(done){
-         chai.request(server)
-          .put('/travellers')         // note the PUT method
-          .send({surname: 'Polo'})    // attach the payload, encoded as JSON
-          .end(function(err, res){    // Send the request. Pass a Node callback
+    // ### EXAMPLE ###
+    suite("PUT /travellers", function () {
+      test('#example - responds with appropriate JSON data when sending {surname: "Polo"}', function (done) {
+        chai
+          .request(server)
+          .put("/travellers") // note the PUT method
+          .send({ surname: "Polo" }) // attach the payload, encoded as JSON
+          .end(function (err, res) {
+            // Send the request. Pass a Node callback
```

Comment:
```suggestion
    suite("PUT /travellers", function () {
      test('#example - responds with appropriate JSON data when sending {surname: "Polo"}', function (done) {
        chai
          .request(server)
          .put("/travellers")
          .send({ surname: "Polo" })
          .end(function (err, res) {
```

There's a lot of comments in this file, I feel like most or all of them should be removed and, if it's necessary info, moved over to /learn.

Reply:
That is a fair enough assessment. I left them, because the code does not get changed by the camper, so there is nothing to go wrong (maintenance-wise).

I would not suggest removing the comments, unless the whole code examples are pasted into `/learn`; it is difficult to explain large sections of code with words, and it is even more difficult to follow an explanation of large sections of code.

So, shall I copy-pasta this as a minimum to `/learn`, then remove the comments from this boilerplate?
```javascript
chai
          .request(server)
          .put("/travellers") // note the PUT method
          .send({ surname: "Polo" }) // attach the payload, encoded as JSON
          .end(function (err, res) {
            // Send the request. Pass a Node callback
```

Reply:
Yea, putting all those code blocks in there might be a little much. I was thinking to describe it on learn, but that's kind of tough. Just leave it for now, we can add it as-is if we need to, it's certainly an improvement.